core/metrics-core-impl/src/main/java/org/hawkular/metrics/core/impl/cassandra/MetricsServiceCassandra.java

```diff
@@ -701,9 +700,9 @@ private void dropKeyspace(String keyspace) {
                     return Futures.immediateFuture(results);
                 }
                 List<ResultSetFuture> insertFutures = new ArrayList<>();
-                tags.forEach((k, v) -> dataAccess.insertAvailabilityTag(k, v.orElse(""), data));
+                tags.forEach((k, v) -> dataAccess.insertAvailabilityTag(k, v, data));
                 for (Availability a : data) {
-                    insertFutures.add(dataAccess.updateDataWithTag(a, MetricUtils.flattenTags(tags)));
+                    insertFutures.add(dataAccess.updateDataWithTag(a, tags));
```

Comment:
I haven't looked at the details, why all these changes in the tags model? (It used to be `Map<String, Optional<String>>` everywhere)